### PR TITLE
[C] support more color format in ColorTypeConverter

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

[C] support more color format in ColorTypeConverter

### Bugs Fixed ###

/

### API Changes ###

/

### Behavioral Changes ###

/

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core/ColorTypeConverter.cs [hunk 1]
```diff
@@ -6,171 +6,253 @@ namespace Xamarin.Forms
...
-					case "Tomato": return Color.Tomato;
-					case "Transparent": return Color.Transparent;
-					case "Turquoise": return Color.Turquoise;
-					case "Violet": return Color.Violet;
-					case "Wheat": return Color.Wheat;
-					case "White": return Color.White;
-					case "WhiteSmoke": return Color.WhiteSmoke;
-					case "Yellow": return Color.Yellow;
-					case "YellowGreen": return Color.YellowGreen;
+					switch (color.ToLowerInvariant()) {
```
Comment (samhouts) [inline Xamarin.Forms.Core/ColorTypeConverter.cs]: I bet this was fun.

File: Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs [hunk 2]
```diff
@@ -245,12 +245,23 @@ public void TestColorTypeConverter ()
...
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("rgb(0%,0%, 110%)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("hsl(240,100%, 50%)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("hsl(240,110%, 50%)"));
+			Assert.AreEqual(Color.Blue.MultiplyAlpha(.8), converter.ConvertFromInvariantString("hsla(240,100%, 50%, .8)"));
+			Assert.AreEqual(Color.Default, converter.ConvertFromInvariantString("Color.Default"));
+			Assert.AreEqual(Color.Accent, converter.ConvertFromInvariantString("Accent"));
+			var hotpink = Color.FromHex("#FF69B4");
 			Color.Accent = hotpink;
 			Assert.AreEqual (Color.Accent, converter.ConvertFromInvariantString ("Accent"));
 			Assert.Throws<InvalidOperationException> (() => converter.ConvertFromInvariantString (""));
```
Comment (samhouts) [inline Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs]: Maybe also add some `Assert.Throws` for parentheses violations (e.g., ColorTypeConverter, L28).

File: Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs [hunk 3]
```diff
@@ -245,12 +245,23 @@ public void TestColorTypeConverter ()
...
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("Color.Blue"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("Blue"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("blue"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("#0000ff"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("#00f"));
+			Assert.AreEqual(Color.Blue.MultiplyAlpha(2.0/3.0), converter.ConvertFromInvariantString("#a00f"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("rgb(0,0, 255)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("rgb(0,0, 300)"));
+			Assert.AreEqual(Color.Blue, converter.ConvertFromInvariantString("rgb(0,0, 300)"));
+			Assert.AreEqual(Color.Blue.MultiplyAlpha(.8), converter.ConvertFromInvariantString("rgba(0%,0%, 100%, .8)"));
```
Comment (samhouts) [inline Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs]: This one is failing.. :(

File: Xamarin.Forms.Core/ColorTypeConverter.cs (hunk 1 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/ColorTypeConverter.cs]: @samhouts I bet it was fun to review, trying to find the one place an uppercase was left, or the one time I mixed `gray` with `blue`, accidentally creating the `glue` color.

File: Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs (hunk 3 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs]: WORKSFORME... investigating...

File: Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs (hunk 2 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core.UnitTests/ColorUnitTests.cs]: I'm still unsure about how this should behave on error. While parsing Xaml, we throw, but while parsing ~~css~~ any-other-markup, we might want to revert to default.

adding a test nonetheless

**Pull request merged**